### PR TITLE
br: Support backup replica read

### PR DESCRIPTION
**Pull request by v01dstar**
Signed-off-by: Yang Zhang <yang.zhang@pingcap.com>

### What problem does this PR solve?

Issue Number: ref #40898 40898

Problem Summary:

### What is changed and how it works?

* Allow users use replica read in backup through the argument.
* In backup pushdown, TiKV will check whether the peer hosted in the store meets the role requirements.
* In fine grained backup, the backup client will check which peer meets the requirements.

Corresponding TiKV PR: https://github.com/tikv/tikv/pull/13975

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [x] Contains variable changes
- [x] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Support replica read in backup
```

### CODE REVIEW

File: br/pkg/backup/client.go [hunk 1]
```diff
@@ -859,8 +874,8 @@ func (bc *Client) BackupRange(
 			req.EndKey = progressRange.Incomplete[0].EndKey
 		}
 
-		push := newPushDown(bc.mgr, len(allStores))
-		err = push.pushBackup(ctx, req, progressRange, allStores, bc.checkpointRunner, progressCallBack)
+		push := newPushDown(bc.mgr, len(targetStores))
+		err = push.pushBackup(ctx, req, progressRange, targetStores, bc.checkpointRunner, progressCallBack)
```
Comment (YuJuncen) [inline br/pkg/backup/client.go:878]: When the `replicaLabels` is an empty map, will `targetStores` be `nil` here? If so, all the backup procedure would be executed in the fine-grained backup, which is almost unacceptable.

**Review by YuJuncen — COMMENTED** [review]
rest lgtm

File: br/pkg/backup/client.go (hunk 1 above)
Comment (v01dstar) [reply br/pkg/backup/client.go:882]: Thank you for catching this, updated. PTAL

**Review by YuJuncen — APPROVED** [review]

File: br/pkg/task/backup.go [hunk 2]
```diff
@@ -77,14 +78,15 @@ type CompressionConfig struct {
...
-	UseCheckpoint    bool          `json:"use-checkpoint" toml:"use-checkpoint"`
+	TimeAgo          time.Duration     `json:"time-ago" toml:"time-ago"`
+	BackupTS         uint64            `json:"backup-ts" toml:"backup-ts"`
+	LastBackupTS     uint64            `json:"last-backup-ts" toml:"last-backup-ts"`
+	GCTTL            int64             `json:"gc-ttl" toml:"gc-ttl"`
+	RemoveSchedulers bool              `json:"remove-schedulers" toml:"remove-schedulers"`
+	IgnoreStats      bool              `json:"ignore-stats" toml:"ignore-stats"`
+	UseBackupMetaV2  bool              `json:"use-backupmeta-v2"`
+	UseCheckpoint    bool              `json:"use-checkpoint" toml:"use-checkpoint"`
+	ReplicaReadLabel map[string]string `json:"backup-replica-read-label" toml:"backup-replica-read-label"`
```
Comment (3pointer) [inline br/pkg/task/backup.go:89]:
```suggestion
	ReplicaReadLabel map[string]string `json:"replica-read-label" toml:"replica-read-label"`
```

**Review by 3pointer — APPROVED** [review]
rest LGTM

**Pull request merged**